src/main/webapp/app/exercises/programming/manage/services/programming-exercise.service.ts

```diff
@@ -119,6 +121,7 @@ export class ProgrammingExerciseService {
     update(programmingExercise: ProgrammingExercise, req?: any): Observable<EntityResponseType> {
         const options = createRequestOption(req);
         const copy = this.convertDataFromClient(programmingExercise);
+        this.exerciseService.setBonusPointsConstrainedByIncludedInOverallScore(copy);
```

Comment:
I think it would be easier to understand if you wrote `copy = this.exerciseService.setBonusPointsConstrainedByIncludedInOverallScore(copy)` because at the first glance I thought you were setting some internal state in the exercise service.

Reply:
I've improved the readability in the commit: 64ccda3c8ba7055a447cff6b1ccb7e824b51d6a2